llvm/include/llvm/Frontend/Offloading/SYCLOffloadWrapper.h

```diff
@@ -30,7 +31,20 @@ enum class SYCLBinaryImageFormat {
 };
 
 struct SYCLImage {
```

Comment:
Is it possible to reuse or extend llvm::OffloadBinary::OffloadingImage data structure for this? ImageKind is the same as Format here. OffloadKind is OFK_SYCL. Entries and Properties could be moved into StringData.
https://llvm.org/docs/doxygen/structllvm_1_1object_1_1OffloadBinary_1_1OffloadingImage.html

I am Ok with adding a TODO here and doing this in part 2 of changes.

Thanks

Reply:
This is a good idea. I will think through it after this patch.